### DISCUSSION TITLE
How to tell if chain is changed from wallet or from `switchChain`?

### DISCUSSION THREAD

**Discussion by hungdoansy** [wevm/wagmi#3824] [Q&A]
Hi guys,

I need your help with this issue.

# Story

1. I have a webapp and I use Rainbowkit for wallet connection
2. My webapp supports only several chains: Ethereum, Base and zkSync. Only 3
3. When users change network in their wallet, the webapp will not react and retain what network it currently is.
4. If the network is changed by user using Rainbowkit chain modal, I'd like to update `config.chainId` to the new one.

# More details

Chain can be changed by 2 ways:
1. User changes it in their wallet
2. `switchChain` within the webapp

In case 1, I don't want our app to react to it. If the webapp's in Ethereum, it will still be Ethereum no matter what they try to change in the their wallet.
In case 2, it is the user's intention to change the network so I'd like the webapp to save this latest chain.

In summary, I only want the webapp to change chainId (in the config) only if this is done within our webapp.

# What I did

1. I set `syncConnectedChain = false` so that now there's no connection between `account.chainId` and `config.chainId`
2. I need to write my own sync logic.
3. **But**, I have no way to tell if network is changed by connector, or by `switchChain`.

Thanks in advance.

**Comment by hungdoansy** [comment]
Hi @jxom can you share some thoughts?
